### DISCUSSION TITLE
TaskFlow API with parallel tasks?

### DISCUSSION THREAD

**Discussion by csm10495** [apache/airflow#20643] [Q&A] [answered]
Hey folks,

Is there a way to use the TaskFlow API and mark tasks to run at the same time?

Like say I have TaskA and TaskB that are both a bit long and I'd like to run them at the same time... can that be done using TaskFlow?

**Comment by csm10495** [comment]
Or does it just guess based off the usage of xcom outputs the order of tasks?

**Comment by potiuk** [comment, marked as answer]
if you do not make task depend on each other, they will run in parallel.

In the [TaskFlow tutorial](https://airflow.apache.org/docs/apache-airflow/stable/tutorial_taskflow_api.html#using-the-taskflow-api-with-docker-or-virtual-environments) for example the dependency between extract and transform task is established by this:

```
order_data = extract()
order_summary = transform(order_data)
```

But if transform is not set to consume extract's data, they would run in parallel.
